horovod/common/util.py

```diff
@@ -251,3 +253,11 @@ def split_list(l, n):
     """
     d, r = divmod(len(l), n)
     return [l[i * d + min(i, r):(i + 1) * d + min(i + 1, r)] for i in range(n)]
+
+
+def check_framework_version(name, version):
+    file_path = os.path.abspath(os.path.dirname(os.path.abspath(__file__)))+"/../metadata.json"
+    with open(file_path) as f:
+        install_version = json.load(f)
+        if not install_version[name] == version:
+            raise HorovodVersionMismatch(install_version[name])
```

Comment:
We should avoid raising a`RuntimeWarning`.  Instead, you can do something like:

```
warnings.warn(f'Framework {name} installed with version {installed_version} but found version {version}.  This can result in unexpected behavior including runtime errors.  Reinstall Horovod using `pip install --no-cache-dir` to build with the new version.')
```

Something like that.